build/deploy/cockroach.sh

```diff
@@ -1,10 +1,232 @@
+                      -e "GRANT ALL ON DATABASE $COCKROACH_DATABASE TO $COCKROACH_USER" \
+                      -e "GRANT admin TO $COCKROACH_USER " )
+
+    # Check if the given command contains the `--insecure` flag.
+    # If not, add the `--certs-dir=certs` to init_env_query.
+    if [[ "$*" =~ $INSECURE_MODE_REGEX ]]; then
+              init_env_query+=( --insecure )
+              else
+              init_env_query+=( "--certs-dir=$CERTS_DIR" )
+    fi
+
+    run_sql_query "${init_env_query[@]}"
+}
+
+# start_init_node is to start the single node for the initialization.
+start_init_node() {
+  echo "Starting node for the initialization process. This could take couple seconds"
+  rm -f server_fifo; mkfifo server_fifo
+  local start_node_query=( $COCKROACH_ENTRYPOINT start-single-node \
+                           --background --listening-url-file=server_fifo \
+                           --pid-file=server_pid )
+
+  # Check if the given command contains the `--insecure` flag.
+  # If not, add the `--certs-dir=certs` to init_env_query.
+  if [[ "$*" =~ $INSECURE_MODE_REGEX ]]; then
+    start_node_query+=( --insecure )
+    else
+    start_node_query+=( "--certs-dir=$CERTS_DIR" )
+  fi
+
+  # Get the given listening address.
+  [[ "$*" =~ $LISTEN_ADDR_REGEX ]] && start_node_query+=( $BASH_REMATCH )
+
+  # Start the node and run in the background.
+  "${start_node_query[@]}" &
+
+  # Set a 5-minute timeout for the single node starting up.
+  timeout 3000s cat server_fifo>server.url
+  exit_status=$?
+  if [[ $exit_status -eq 124 ]]; then
+      echo >&2 "error: timeout for cockroach init process"
+      exit 1
+  fi
+}
+
+# stop_init_node is to stop the single node for the initialization.
+stop_init_node() {
+    kill $(cat server_pid)
+    while ! kill -0 $(cat server_pid); do
+        echo "finishing cockroach init process"
+        sleep 3
+    done
+
+    sleep 5s
+    echo "cockroach init process finished, restart the server now"
+}
+
+_main() {
+  if [ "${1-}" = "shell" ]; then
+    shift
+    exec /bin/sh "$@"
+  else
+    shopt -s nocasematch
+    if [[ "$*" =~ $START_SINGLE_NODE_REGEX ]]; then
+      echo "start init process"
+      STRING_TO_REMOVE=`echo ${BASH_REMATCH//[[:blank:]]/}`
+      set -- ${*//$STRING_TO_REMOVE/}
+
+      # If the cockroach-data directory is empty.
+      if ! [ "$(ls -A cockroach-data)" ]; then
+        analyze_command "$*"
+
+        if [[ "$*" =~ $CERTSDIR_REGEX ]]; then
+          setup_certs_dir
+        fi
+
+        setup_env
+        start_init_node "$*"
+        setup_db "$*"
+        process_init_files /docker-entrypoint-initdb.d/*
+        stop_init_node
+      fi
+      # restart server.
+      exec $COCKROACH_ENTRYPOINT start-single-node "$@"
+    else
+      if [[ "$*" =~ $CERTSDIR_REGEX ]] && ! [[ "$*" =~ $URL_REGEX ]]; then
+        echo >&2 "error: in secure mode, --url must be specified. \
+        check https://www.cockroachlabs.com/docs/v21.1/connection-parameters#connect-using-a-url"
```

Comment:
drive by comment: the `v21.1` should either match the version we're backporting to, or point to `stable`